# LO22 – TP1 : Blocs digrammes de Fiabilité 

In [1]:
from fiabilipym import Component, System
import plotly.graph_objects as go
import plotly.io as pio
pio.renderers.default = 'iframe'

## 1. Architecture 1

**1.1. Modéliser l’architecture 1 du réseau de périphérique de stockage par un bloc
diagramme de fiabilité.**

![Architecture 1](A1.png)

In [2]:
# Components

hba1 = Component("HBA1", 1.25 * 10**-6, 7e-3)
hba2 = Component("HBA2", 1.25 * 10**-6, 7e-3)
linka = Component("LINKA", 2.52 * 10**-6, 3e-3)
linkb = Component("LINKB", 2.52 * 10**-6, 3e-3)
hub = Component("HUB", 1.71 * 10**-6, 2e-3)
linkc = Component("LINKC", 2.52 * 10**-6, 3e-3)
linkd = Component("LINKD", 1.71 * 10**-6, 3e-3)
diska = Component("DISKA", 10**-5, 5e-4)
diskb = Component("DISKB", 10**-5, 5e-4)


# System

s1 = System()
s1["E"] = [hba1, hba2]

s1[hba1] = linka
s1[hba2] = linkb

s1[linka] = s1[linkb] = hub

s1[hub] = [linkc, linkd]

s1[linkc] = diska
s1[linkd] = diskb

s1[diska] = s1[diskb] = "S"

**1.2. Tracer les courbes donnant la disponibilité et la fiabilité du réseau de périphériquede
stockage sur une durée de 2 ans.**

In [3]:
timerange = range(0, 2 * 365 * 24, 1)

t = [t for t in timerange]
reliability = [float(s1.reliability(t)) for t in timerange]
availability = [float(s1.availability(t)) for t in timerange]

In [4]:
len(reliability)

17520

In [5]:
fig = go.Figure()
fig.update_layout(title="Courbes de disponibilité et fiabilté pour le système 1")
fig.update_xaxes(title_text="Temps(h)")

fig.add_trace(go.Scatter(x=t, y=reliability, name="Reliability"))
fig.add_trace(go.Scatter(x=t, y=availability, name="Availability"))
fig.show()

**1.3. Donner la valeur de la fiabilité et de la disponibilité asymptotique du système (à tinfini).**

In [6]:
print(reliability[-1])
print(availability[-1])

0.9312165832370191
0.9987331404588206


Valeur de la fiabilité asymptotique : $0.931$

Valeur de la disponibilité asymptotique : $0.999$

## 2. Architecture 2

**2.1. Modéliser l’architecture 2 du réseau de périphérique de stockage par un bloc
diagramme de fiabilité.**

![Architecture 2](A2.png)

In [7]:
# Components

hba1 = Component("HBA1", 1.25 * 10**-6, 7e-3)
hba2 = Component("HBA2", 1.25 * 10**-6, 7e-3)
linka = Component("LINKA", 2.52 * 10**-6, 3e-3)
linkb = Component("LINKB", 2.52 * 10**-6, 3e-3)
hub1 = Component("HUB1", 1.71 * 10**-6, 2e-3)
hub2 = Component("HUB2", 1.71 * 10**-6, 2e-3)
linkc = Component("LINKC", 2.52 * 10**-6, 3e-3)
linkd = Component("LINKD", 1.71 * 10**-6, 3e-3)
diska = Component("DISKA", 10**-5, 5e-4)
diskb = Component("DISKB", 10**-5, 5e-4)


# System

s2 = System()

s2["E"] = [hba1, hba2]

s2[hba1] = linka
s2[hba2] = linkb

s2[linka] = hub1
s2[linkb] = hub2

s2[hub1] = linkc
s2[hub2] = linkd

s2[linkc] = diska
s2[linkd] = diskb

s2[diska] = s2[diskb] = "S"

**2.2. Tracer les courbes donnant la fiabilité et la disponibilité du réseau de périphérique de
stockage sur une durée de 2 ans.**

In [8]:
timerange = range(0, 2 * 365 * 24, 1)
reliability = [float(s2.reliability(t)) for t in timerange]
availability = [float(s2.availability(t)) for t in timerange]

In [9]:
fig = go.Figure()
fig.update_layout(title="Courbes de disponibilité et fiabilté pour le système 2")
fig.update_xaxes(title_text="Temps(h)")

fig.add_trace(go.Scatter(x=t, y=reliability, name="Reliability"))
fig.add_trace(go.Scatter(x=t, y=availability, name="Availability"))
fig.show()

**2.3  Donner la valeur de la disponibilité asymptotique du système (à tinfini).**

In [10]:
print(availability[-1])
print(reliability[-1])

0.9995103228565482
0.929670915797872


Valeur de la fiabilité asymptotique : $0.995$

Valeur de la disponibilité asymptotique : $0.923$

## 3. Comparaison des architectures

**3.1. Comparer les disponibilités asymptotiques des deux architectures et conclure par rapport à
l’intérêt de l’utilisation de la redondance au niveau des architectures.**

**3.2. Proposer une troisième architecture ayant les mêmes composants que l’architecture 2
mais une disponibilité supérieure.**

## **4. Facteurs d’importance**

**4.1. Calculer le facteur d’importance de Birnbaum des composants de l’architecture 1.**

**4.2. Conclure quand au(x) composant(s) qu’il faut ajouter en redondance pour améliorer la
fiabilité du réseau de périphérique de stockage.**